src/main/scripts/org/reaktivity/specification/socks/rfc1928/forward/client.connect.send.data/server.rpt

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright 2016-2017 The Reaktivity Project
+#
+# The Reaktivity Project licenses this file to you under the Apache License,
+# version 2.0 (the "License"); you may not use this file except in compliance
+# with the License. You may obtain a copy of the License at:
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+
+property newServerAcceptRef ${nuklei:newReferenceId()}
+property newClientConnectRef ${newServerAcceptRef}
+
+property serverAccept "nukleus://socks/streams/source"
+
+accept ${serverAccept}
+  option nukleus:route ${newClientConnectRef}
+  option nukleus:window 65536
+  option nukleus:transmission "duplex"
+accepted
+connected
+
+## Read the request bytes for connecting with no-authentication
+read [0x05 0x01 0x00]
+
+## Write the response, expect success
+write [0x05 0x00]
+
+## Read CONNECT request
+read [0x05] # version: SOCKS v5
+read [0x01] # CONNECT command
+read [0x00] # RSV field
+read [0x03] # Address Type domain name
+read [0x0B] # Length of the FQDN
+read "example.com" # The FQDN
+read [0x1F 0x90]   # The destination port
+
+## Write CONNECT response
+write [0x05] # version: SOCKS v5
+write [0x00] # Succeeded
+write [0x00] # RSV field
+write [0x01] # Address Type domain name
+write [0x7F 0x00 0x00 0x01] # The Bound Server Address
+write [0xEA 0xDB]           # The Bound Server Port
```

Comment:
Isn't this suppose to be the ephemeral port?  How do we get this?

Reply:
On the spec project it is sent back by `server.rpt` in the same directory. 

In the implementation test it will be taken from `.../specification/nukleus/socks/streams/forward/client.connect.send.data/server.rpt`. This is because the socks nukleus, acting as `server`, will use the `TcpBeginExFW` of the response from target to construct the Socks Connect Response. 

When acting as `client`, it receives this information as an already encoded socks message but makes no use of it (as already discussed).

Please see below a diagram of the `server` and `client` behaviors in `forward` mode. The example routing might need more precision, but it should not contain errors affecting the socks spec.